### PR TITLE
update to the app structure section

### PR DESCRIPTION
**Pull request by kristojorg**
I thought this was a more strightforward order for describing things,
and it puts one of the most unique parts of Gatsby 1.0 right at the
top.

### CODE REVIEW

**Comment by kristojorg** [comment]
As I started to put more and more thought into this it made more sense to instead take what I'm learning from reading this post and make an effort at documentation instead. Let me know what you think of this reformulation however.

**Comment by KyleAMathews** [comment]
This is really nice! I think this is much clearer and more interesting. Thanks for the refactor!

And on documentation — feel free to copy anything and everything from these blog posts for doc pages. I think lots of duplication of ideas and languages is perfectly good. I see blog posts as a cheaper way to experiment with ways of explaining things.

File: docs/blog/2017-03-08-gatsbygram-case-study/index.md [hunk 1]
```diff
@@ -109,83 +109,24 @@ Stop wasting time and build something.
...
+    blog1.md
 ```
+would be transformed to
+```
+my-site/
+  index.html
+  blogs/
+    blog1.html
+```
+This is fine at first, but can be limiting. For example, in gatsbygram, we have a json data blob scraped from the instagram api. From this we want to generate a page for each image. We couldn't do this with a typical static site generator, but Gatsby lets you define routes programatically through the `createPages` api.
```
Comment (KyleAMathews) [inline docs/blog/2017-03-08-gatsbygram-case-study/index.md]: Capitalize gatsbygram and instagram throughout.

File: docs/blog/2017-03-08-gatsbygram-case-study/index.md [hunk 2]
```diff
@@ -252,13 +193,13 @@ exports.createPages = ({ args }) => (
...
-The optional context data you specify when creating pages is
-automatically passed in as a prop to the component. They're also passed
-as [GraphQL variables](http://graphql.org/learn/queries/#variables) so
-you can easily write dynamic queries for additional information.
+This allows us to create pages from any data, we just have to have the data accessible through our graphql schema first... more on that later.
+
+## Using templates
 
-This is what Gatsbygram's post template component looks like:
+Gatsby uses standard React.js components to render static pages. When you define a page in the `createPages` api, you also define a component to build it with. Those components, usually called templates get reused with different data to generate different pages. Usually, we will pass some simple information about what page we are on to the template component, and then the template will make a `graphql` query using that information to get the data it needs.
```
Comment (KyleAMathews) [inline docs/blog/2017-03-08-gatsbygram-case-study/index.md]: change "render static pages" to just "render pages" as pages can of course be highly dynamic even completely dynamic e.g. don't render anything until the client.

Comment (KyleAMathews) [inline docs/blog/2017-03-08-gatsbygram-case-study/index.md]: Add comma after "usually called templates"

Comment (KyleAMathews) [inline docs/blog/2017-03-08-gatsbygram-case-study/index.md]: Don't need commas in "Usually, we will..." sentence. That probably should be broken into two sentences remove the "and" in the middle.

File: docs/blog/2017-03-08-gatsbygram-case-study/index.md [hunk 3]
```diff
@@ -317,14 +258,19 @@ export const pageQuery = `
 `
 ```
 
-### React.js pages
+## Creating one-off pages
 
-Gatsby lets you build pages from individual React.js components. Like
-template components, you can add GraphQL queries to query for data.
+In addition to creating a page for every one of our instagram images, we also want to make an index page that shows off our insta grid. To do so, Gatsby lets us define pages using the file system, just like your standard static site generators:
```
Comment (KyleAMathews) [inline docs/blog/2017-03-08-gatsbygram-case-study/index.md]: "insta grid" typo.

File: docs/blog/2017-03-08-gatsbygram-case-study/index.md [hunk 4]
```diff
@@ -317,14 +258,19 @@ export const pageQuery = `
...
+```
+pages/
+  index.js
+  about.js
+```
+
+These react components can query our graphql schema for data, and will each be generated into their own pages at `gatsbygram.com/` and `gatsbygram.com/about`.
```
Comment (KyleAMathews) [inline docs/blog/2017-03-08-gatsbygram-case-study/index.md]: React should be capitalized throughout

Comment (KyleAMathews) [inline docs/blog/2017-03-08-gatsbygram-case-study/index.md]: "our graphql schema" a bit awkward — perhaps "query the Gatsbygram GraphQL schema" or just "query their site's GraphQL schema".

Comment (KyleAMathews) [inline docs/blog/2017-03-08-gatsbygram-case-study/index.md]: GraphQL proper capitalization

Comment (KyleAMathews) [inline docs/blog/2017-03-08-gatsbygram-case-study/index.md]: "will each be generated" — a bit awkward

**Comment by KyleAMathews** [comment]
Hey, trying to get the post out this morning. Just going to merge and edit locally. Thanks again!

**Comment by kristojorg** [comment]
Yea these updates look good to me!

**Comment by jlengstorf** [comment]
Hiya @kristojorg! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**